src/web/lib/storage.js

```diff
@@ -17,16 +17,32 @@ const generateThemeKey = () =>
   `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
 
 function putTheme(key, theme) {
-  log("putTheme", key, theme);
-  const storageKey = themeStorageKey(key);
-  localStorage.setItem(
-    storageKey,
-    JSON.stringify({
-      theme,
-      modified: Date.now()
-    })
-  );
-  notifySelfForStorage(storageKey);
+  checkDuplicateTheme(theme, (isThemeDuplicate) => {
+    if (!isThemeDuplicate) {
```

Comment:
Since `localStorage.setItem` is syncronous, you don't need a callback here. You can just return the result from `isThemeDuplicate`